pynamodb/attributes.py

```diff
@@ -807,6 +825,9 @@ def validate(self):
         return all(self.is_correctly_typed(k, v) for k, v in self.get_attributes().items())
 
     def serialize(self, values):
+        if isinstance(values, type(self)) and not values.is_raw():
```

Comment:
Is this because ListAttribute creates an anonymous `element_type` attribute instance to serialize?

Reply:
ListAttributes serializes based on the value itself if `isinstance` passes.

Reply:
Well now it does, just validating why this logic was needed.

Reply:
Right, so before that change ListAttribute also had the same behavior where it would only serialize elements based on the attributes in the declared type.

Supporting polymorphism requires that we serialize all elements so we don't drop any data. Now until discriminator map support lands we won't be able to deserialize all of the data on disk since we won't know which subtype to instantiate.